panic with go btime 1

```
position fen 7r/5R1P/4ppK1/8/6P1/6k1/8/8 b - - 5 44
go btime 1
panic: close of closed channel

goroutine 15 [running]:
github.com/paulsonkoly/chess-3/uci.(*Engine).handleGo(0x14000100450, {0x14000106010, 0x2, 0x0?})
        /Users/phaul/go/src/github.com/paulsonkoly/chess-3/uci/uci.go:329 +0x56c
github.com/paulsonkoly/chess-3/uci.(*Engine).handleCommand(0x14000100450, {0x14000090010?, 0x0?})
        /Users/phaul/go/src/github.com/paulsonkoly/chess-3/uci/uci.go:120 +0x84
github.com/paulsonkoly/chess-3/uci.(*Engine).handleInput(...)
        /Users/phaul/go/src/github.com/paulsonkoly/chess-3/uci/uci.go:97
github.com/paulsonkoly/chess-3/uci.(*Engine).Run.func2()
        /Users/phaul/go/src/github.com/paulsonkoly/chess-3/uci/uci.go:61 +0x50
created by github.com/paulsonkoly/chess-3/uci.(*Engine).Run in goroutine 1
        /Users/phaul/go/src/github.com/paulsonkoly/chess-3/uci/uci.go:60 +0x1d4
```

--- a/uci/uci.go
+++ b/uci/uci.go
@@ -273,6 +273,11 @@ func (tc timeControl) hardLimit(stm Color) int64 {
 	if stm == Black && tc.btime > 0 {
 		timeLeft = tc.btime
 	}
+	
+	if timeLeft <= TimeSafetyMargin {
+		// we are losing on time anyway, but at least allocate time
+		return timeLeft
+	}
 
 	return Clamp(3*tc.softLimit(stm), TimeSafetyMargin, timeLeft-TimeSafetyMargin)
 }
@@ -325,12 +330,16 @@ func (e *Engine) handleGo(args []string) {
 			finished = true
 
 		case <-time.After(time.Duration(tc.hardLimit(stm)) * time.Millisecond):
-			stopped = true
-			close(e.SST.Stop)
+			if !stopped {
+				stopped = true
+				close(e.SST.Stop)
+			}
 
 		case <-e.stop:
-			stopped = true
-			close(e.SST.Stop)
+			if !stopped {
+				stopped = true
+				close(e.SST.Stop)
+			}
 		}
 	}
 

